### ISSUE TITLE
sql: CHECK constraints are not dropped when their columns are

### ISSUE THREAD

**Issue by benesch** [cockroachdb/cockroach#20630]
CockroachDB:

```
> create table a (a int, b int, check (a < b));
> alter table a drop column b;
> insert into a values (1);
pq: column name "b" not found
```

PostgreSQL:

```
> create table a (a int, b int, check (a < b));
> alter table a drop column b;
> insert into a values (1);
INSERT 0 1
```

PostgreSQL is auto-dropping the constraint when the column is dropped. We should do the same.

@vivekmenezes can you triage?

**Comment by vivekmenezes** [comment]
Leaving it assigned to self for now.

**Comment by dt** [comment]
Huh, I'm surprised you can drop the column (in pg) while it is depended on by the multi-column constraint -- i'd have expected an error from doing that (at least without cascade).

e.g. let's say you have some products, e.g. `item, price, discount, ...` and you add a constraint to make sure you never accidentally make items have net negative prices: `CHECK (price >= discount)`. Later you add an additional  `member_discount`, and update your constraint to be (price >= discount + member_discount)`... not if you later drop `member_discount`, your constraint on the other two columns is just silently gone? That seems scary to me.

That said, I've learned by now just to do what postgres does and not argue, but still, ugh.

**Comment by ZacharyThomas** [comment]
Going to look at this if it's as simple as dropping check constraints on drop column statements and @vivekmenezes isn't looking at it

**Comment by vivekmenezes** [comment]
Thanks!

**Comment by nvanbenschoten** [comment]
Fixed by #21203.
